pkg/api/types.go

```diff
@@ -551,6 +553,20 @@ type RBDVolumeSource struct {
 	ReadOnly bool `json:"readOnly,omitempty"`
 }
 
+// MetadataVolumeSource represents a volume containing metadata about a pod.
+type MetadataVolumeSource struct {
+	// Items is a list of metadata file name
+	Items []MetadataFile `json:"items,omitempty"`
+}
+
+// MetadataFile represents information to create the file containing the pod field
+type MetadataFile struct {
+	// Required: Name is the name of the file
+	Name string `json:"name,omitempty"`
```

Comment:
Is this a relative path, or just a filename, or...?

Are there any character-set restrictions? Is there any reason it should have different restrictions than secret map keys?
https://github.com/kubernetes/kubernetes/blob/master/pkg/api/v1/types.go#L1936

Also, required fields should not be tagged with `omitempty`, which implies optional.